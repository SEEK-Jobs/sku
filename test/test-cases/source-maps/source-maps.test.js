const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');

describe('source-maps', () => {
  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', __dirname);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(`${__dirname}/dist`);
      expect(files).toMatchSnapshot();
    });
  });
});
