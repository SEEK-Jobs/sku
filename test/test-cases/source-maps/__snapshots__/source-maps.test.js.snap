// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source-maps build should generate the expected files 1`] = `
Object {
  "index.html": <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4005/runtime-e435f5198c56ab503cd3.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4005/vendors~main-729a6d85c20c37c219a5.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4005/main-37c86d0af6edb1295ff4.js"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div data-reactroot>
        Hello source maps
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4005/runtime-e435f5198c56ab503cd3.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4005/vendors~main-729a6d85c20c37c219a5.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4005/main-37c86d0af6edb1295ff4.js"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "main-37c86d0af6edb1295ff4.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main-37c86d0af6edb1295ff4.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-e435f5198c56ab503cd3.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-e435f5198c56ab503cd3.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-729a6d85c20c37c219a5.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-729a6d85c20c37c219a5.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
